### PR TITLE
Bump ics23 dep to 0.11.3, bump ibc-types version to 0.14.1

### PR DESCRIPTION
**Pull request by zbuc**

### CODE REVIEW

File: ci/no-std-check/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ ibc-types = { path = "../../crates/ibc-types", default-features = false, feature
   "serde",
   "mocks-no-std",
 ] }
-ibc-proto = { version = "0.42.0", default-features = false }
+ibc-proto = { version = "0.42.2", default-features = false }
```
Comment (conorsch) [inline ci/no-std-check/Cargo.toml:12]: Why the change to `ibc-proto` throughout, too? Is that optimistic, or was it required by the `ics23` bump?

**Review by conorsch — APPROVED** [review]
LGTM. We should carefully consider implications for compatibility with tagged Penumbra versions, given concerns in https://github.com/penumbra-zone/penumbra/issues/4682.

File: ci/no-std-check/Cargo.toml (hunk 1 above)
Comment (zbuc) [reply ci/no-std-check/Cargo.toml:12]: ibc-types updated its version of ibc-proto so i brought it into alignment
